### PR TITLE
feat(search): Search Component - Type Ahead

### PR DESCRIPTION
**Pull request by beliayeu**
## Changes
* List of potential keyword phrases appears with the first letter typed as a list below the search box (see screenshot attached) 
* Order is based on frequency and lists no more then 5 items
* Items list # of results 
* Item list is at full width of the search field
* search results do not contain punctuation 
* Responsive behavior should be the same on all formats

### CODE REVIEW

File: examples/test-site/src/components/Search/token.tsx [hunk 1]
```diff
@@ -48,12 +64,34 @@ const responsiveSearchDesign = {
     withIcon('search'),
     addClasses('flex absolute right-0 self-center mr-4'),
   ),
+  Suggestions: withDesign({
```
Comment (dewen) [inline examples/test-site/src/components/Search/token.tsx:67]: Suggestions with design seems duplicated with Line 45-55, suggest merge into a function.

File: packages/bodiless-search/src/SearchClient.ts [hunk 2]
```diff
@@ -46,6 +59,27 @@ class SearchClient implements SearchClientInterface {
     return this.searchEngine.search(filtered);
   };
 
+  suggest = (queryString: string, settings: SuggestionSettings = defaultSuggestSettings) => {
```
Comment (dewen) [inline packages/bodiless-search/src/SearchClient.ts:62]: suggest to define the return type as lunr.TokenSet[]

File: packages/bodiless-search/src/components/Search.tsx [hunk 3]
```diff
@@ -215,6 +229,11 @@ const SearchBoxBase: FC<SearchProps> = ({ components, ...props }) => {
         placeholder={placeholder}
       />
       <SearchButton onClick={onClickHandler} />
+      {
+        queryString !== ''
+        // @ts-ignore
```
Comment (dewen) [inline packages/bodiless-search/src/components/Search.tsx:234]: what type msg is ignored here?

**Review by dewen — DISMISSED** [review]

File: examples/test-site/src/components/Search/token.tsx (hunk 1 above)
Comment (beliayeu) [reply examples/test-site/src/components/Search/token.tsx]: makes sense, merged in 37be9adaf346eb4c100f2a3342e74dd0c57388e5

File: packages/bodiless-search/src/SearchClient.ts (hunk 2 above)
Comment (beliayeu) [reply packages/bodiless-search/src/SearchClient.ts:62]: return type here is array of [Suggestion](https://github.com/johnsonandjohnson/Bodiless-JS/blob/37be9adaf346eb4c100f2a3342e74dd0c57388e5/packages/bodiless-search/src/types.ts#L71) that contains a collection of terms and its frequency. so I think the type is fine here

File: packages/bodiless-search/src/components/Search.tsx (hunk 3 above)
Comment (beliayeu) [reply packages/bodiless-search/src/components/Search.tsx]: enhanced types in 37be9adaf346eb4c100f2a3342e74dd0c57388e5

**Review by dewen — DISMISSED** [review]

**Review by dewen — DISMISSED** [review]

**Review by dewen — DISMISSED** [review]

File: examples/test-site/src/components/Search/token.tsx [hunk 4]
```diff
@@ -36,10 +36,10 @@ const isEven = (item: number) => item % 2 === 0;
 const isOdd = (item: number) => item % 2 === 1;
 
 const withSuggestionsDefaultDesign = withDesign({
-  Wrapper: addClasses('absolute top-full z-50 w-full'),
+  Wrapper: addClasses('absolute top-full z-50 w-full border border-black'),
```
Comment (hvanyo) [inline examples/test-site/src/components/Search/token.tsx:39]: Can you just do this at top of this Search/token.ts file:
asSuggestionsborder = addClasses(...) and then add 
same for hovers, and focus

Comment (beliayeu) [reply examples/test-site/src/components/Search/token.tsx]: fixed in 797013394b7a6bf2f7a7b276ca548ebc86509f1b
@hvanyo, thanks

**Review by dewen — APPROVED** [review]

**Review by konst8 — APPROVED** [review]

**Pull request merged**